src/synapse/dataProvider.js

```diff
@@ -212,18 +225,31 @@ const dataProvider = {
     if (!homeserver || !(resource in resourceMap)) return Promise.reject();
 
     const res = resourceMap[resource];
-
-    const homeserver_url = homeserver + res.path;
-    return Promise.all(
-      params.ids.map(id =>
-        jsonClient(`${homeserver_url}/${id}`, {
-          method: "DELETE",
-          body: JSON.stringify(params.data, filterNullValues),
-        }).then(responses => ({
-          data: responses.map(({ json }) => json),
-        }))
-      )
-    );
+    const homeserver_url = homeserver + res.delete_path;
+
+    if (res.data === "users") {
+      return Promise.all(
+        params.ids.map(id =>
+          jsonClient(`${homeserver_url}/${id}`, {
+            method: "POST",
+            body: JSON.stringify({ erase: true }),
+          }).then(json => ({
+            data: json,
+          }))
+        )
+      );
+    } else if (res.data === "rooms") {
+      return Promise.all(
+        params.ids.map(id =>
+          jsonClient(`${homeserver_url}`, {
+            method: "POST",
+            body: JSON.stringify({ room_id: `${id}` }),
+          }).then(json => ({
+            data: json,
+          }))
+        )
+      );
+    }
```

Comment:
We want to avoid endpoint specific code inside the `dataProvider` as far as possible.

Reply:
I can understand this. The APIs are very different. Do you have any ideas? `params` and `${id}` is not available in `resourceMap`.